pkg/cluster/validate.go

```diff
@@ -6,12 +6,27 @@ package cluster
 import (
 	"context"
 
+	"github.com/Azure/ARO-RP/pkg/api"
 	"github.com/Azure/ARO-RP/pkg/api/validate"
 )
 
 func (m *manager) validateResources(ctx context.Context) error {
+	byoNSG := m.doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG
 	ocDynamicValidator := validate.NewOpenShiftClusterDynamicValidator(
 		m.log, m.env, m.doc.OpenShiftCluster, m.subscriptionDoc, m.fpAuthorizer,
 	)
-	return ocDynamicValidator.Dynamic(ctx)
+	err := ocDynamicValidator.Dynamic(ctx)
+	if err != nil {
+		return err
+	}
+	// If the validation found that it's no longer BYONSG, update the doc
+	// TODO this very like not needed when the API change for BYONSG is introduced
+	if byoNSG != m.doc.OpenShiftCluster.Properties.NetworkProfile.PreconfiguredNSG {
```

Comment:
As mentioned before, let's ask Puneet if we can remove this.  I think we should be consistent with how we have done this in the past on other features.

Reply:
From the meeting last Wed, we'll keep this for now.